flake.nix

```diff
@@ -313,16 +319,31 @@
         }
       ];
 
+      extraHackagePackages = with nixpkgs.lib;
+        (concatLists (map
+          (extraSource: (map
+            (subdir:
+              if subdir == "." then extraSource.src.outPath else "${extraSource.src.outPath}/${subdir}")
+            extraSource.subdirs))
+          extraSources));
+
       projectFor = system:
         let
           pkgs = nixpkgsFor system;
           pkgs' = nixpkgsFor' system;
+          # Usage:
+          extraHackage = inputs.haskell-nix-extra-hackage.mkHackage.${system} extraHackagePackages;
         in
         pkgs.haskell-nix.cabalProject' {
           src = ./.;
-          inherit cabalProjectLocal extraSources;
+          inherit cabalProjectLocal;
           name = "bot-plutus-interface";
           compiler-nix-name = "ghc8107";
+
+          extra-hackages = [ (import extraHackage.hackageNix) ];
+          extra-hackage-tarballs = [{ name = "extraHackage"; index = extraHackage.hackageTarball; }];
```

Comment:
Why do we need both?

Reply:
`extra-hackages` for `callCabalProjectToNix`
`extra-hackage-tarballs` for `mkPkgSet`
see https://github.com/input-output-hk/haskell.nix/pull/535
maybe we could upstream a change to haskell.nix to automatically populate `extra-hackages` from `extra-hackage-tarballs`
that still leaves modules, which are are particular to our case of using a fake hackage